src/base/Context.ts

```diff
@@ -523,6 +523,30 @@ export class Context {
     return invalidDids;
   }
 
+  /**
+   * Returns an Identity when given a did string
+   *
+   * @throws if the Identity does not exist
+   */
+  public async getIdentity(identity: Identity | string): Promise<Identity> {
+    let id;
+    if (typeof identity === 'string') {
+      id = new Identity({ did: identity }, this);
+    } else {
+      id = identity;
+    }
+    const exists = await id.exists();
+
+    if (!exists) {
+      throw new PolymeshError({
+        code: ErrorCode.UnmetPrerequisite,
```

Comment:
I'd say this should be a `DataUnavailable` error, although I can understand the impulse of using `UnmetPrerequisite`